src/libs/downloadUrlFile/index.native.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * rn-fetch-blob is used to download file at given sourceURL in the download folder of device.
+ * Downloaded file will be named as download.ext
+ */
+
+import RNFetchBlob from 'rn-fetch-blob';
+import getFileExtensionFromURL from './helper';
+
+/**
+ * @param {String} sourceURL
+ */
+
+const downloadUrlFile = (sourceURL) => {
+    if (!sourceURL) { return null; }
+
+    // get file extension from sourceURL
+    const fileExtension = getFileExtensionFromURL(sourceURL);
+    if (!fileExtension) { return null; }
+
+    const dirs = RNFetchBlob.fs.dirs;
+    RNFetchBlob.config({
+        // fileCache: true,
+        cacheImage: true,
+        appendExt: fileExtension,
+
+        // set file download config for android
+        addAndroidDownloads: {
+            useDownloadManager: true,
+            notification: true,
+            mediaScannable: true,
+            title: `download.${fileExtension}`,
+            path: `${dirs.DownloadDir}/download.${fileExtension}`,
+        },
+
+        path: `${dirs.DownloadDir}/download.${fileExtension}`,
+    }).fetch('GET', sourceURL, {}).then(() => {
+        // console.log('File saved to ', response.path());
```

Comment:
Does `console.log()` even work in native? Regardless, if you're not doing anything inside `.then()` or `.catch()` then let's get rid of them.

Reply:
Yes, `console.log()` does work in native. I added `.then()` and `.catch()` for testing purpose and you are right that they are not doing anything so i'll remove them.

Reply:
🤦 Well that's embarrassing, of course native has a JS console. Thanks for the reminder.